server/data/data-jmap/src/main/java/org/apache/james/jmap/api/filtering/Rule.java

```diff
@@ -35,8 +35,8 @@
     public static class Id {
 
         public static Id of(String id) {
-            Preconditions.checkNotNull(id, "id should not be null");
-            Preconditions.checkArgument(StringUtils.isNotBlank(id), "id should not be empty");
+            Preconditions.checkArgument(
+                StringUtils.isNotEmpty(id) & StringUtils.isNotBlank(id), "`id` is mandatory");
```

Comment:
& is a bitewise operator. Prefer using the boolean operator &&

Reply:
It a bit tricky, i will separate it.